homeassistant/helpers/device_registry.py

```diff
@@ -312,7 +312,7 @@ def async_get_or_create(
         default_model: str | None | UndefinedType = UNDEFINED,
         default_name: str | None | UndefinedType = UNDEFINED,
         # To disable a device if it gets created
-        disabled_by: DeviceEntryDisabler | None | UndefinedType = UNDEFINED,
```

Comment:
We set it to `UNDEFINED` so we know it is not requested to be changed. This "fix" would re-enable every device when updating it's data.

Reply:
Ok, yeah I almost thought that's the wrong approach.

I guess there must be something wrong in `async_set_disabled_by` (https://github.com/home-assistant/core/blob/2021.12.9/homeassistant/config_entries.py#L987), but I can't really see what could go wrong there.